### PR TITLE
Use Number type in viewBox

### PR DESCRIPTION
**Pull request by kasbah**
The xsd has this wrong I think. W3 spec says "number" and seems to define number as something that can have a floating point (though admitedly I find this spec hard to parse). MDN also says `<number>` and links to [this](https://developer.mozilla.org/en-US/docs/Web/SVG/Content_type#number) which includes floating point. If you merge this I'll send a PR to the xsd repo too. 

My own use-case is defining the viewBox in mm for paper formats defined in inches.

### CODE REVIEW

File: svg/_types.py [hunk 1]
```diff
@@ -41,10 +41,10 @@ def __str__(self) -> str:
...
-    width: int
-    height: int
+    min_x: float
+    min_y: float
+    width: float
+    height: float
 
     def __str__(self) -> str:
-        return f"{self.min_x} {self.min_y} {self.width} {self.height}"
+        return f"{self.min_x:g} {self.min_y:g} {self.width:g} {self.height:g}"
```
Comment (orsinium) [inline svg/_types.py:50]: What's the motivation for the change? `g` turns into scientific notation after 6 signs:

```
'{:g}'.format(10 ** 6)  # '1e+06'
```

Does SVG standard allow scientific notation?

**Comment by orsinium** [comment]
Thank you for your contribution :) If it's a number, you can use the `Number` type from `._types` module. It also includes `Decimal` for the people who want a bit more precision :)

File: svg/_types.py (hunk 1 above)
Comment (kasbah) [reply svg/_types.py:50]: The motivation was actually to prevent something like `0 0 100 100` turning into `0.0 0.0 100.0 100.0` though looking at the spec again (and the MDN one actually being a direct quote of this) it does seem to allow for scientific notation as well. 

https://www.w3.org/TR/SVG11/types.html#DataTypeNumber

>When used in an SVG attribute, a `<number>` is defined differently, to allow numbers with large magnitudes to be specified more concisely:
>
>     number ::= [integer](https://www.w3.org/TR/SVG11/types.html#Integer) ([Ee] [integer](https://www.w3.org/TR/SVG11/types.html#DataTypeInteger))?
>                | [+-]? [0-9]* "." [0-9]+ ([Ee] [integer](https://www.w3.org/TR/SVG11/types.html#DataTypeInteger))?

Comment (orsinium) [reply svg/_types.py:50]: If you don't specify any specifier (as it used to be), `int` will be represented without zeros and `float` with. Maybe, it would be great to drop zeros from float one day, but lets for now keep it consistent with the rest of the code. Currently, all numbers are converted by just calling `str` on them: https://github.com/orsinium-labs/svg.py/blob/master/svg/elements.py#L45. Still, great idea to look into it, thanks 👍🏿

**Comment by orsinium** [comment]
Thank you!

**Comment by kasbah** [comment]
Thanks for reviewing and this great library in general. (In case you are curious I am starting to use it for plotting electronics schematics for [edea.dev](https://blog.edea.dev/introduction/), source-code: https://gitlab.com/edea-dev/edea .)
